### PR TITLE
Add ability to register pipelines through a SCANCODEIO_PIPELINES_DIRS…

### PR DESCRIPTION
**Pull request by tdruez**
… setting #237

Signed-off-by: Thomas Druez <tdruez@nexb.com>

### CODE REVIEW

**Comment by tdruez** [comment]
@pombredanne could you have a look at the changes before the merge?

File: docs/scancodeio-settings.rst [hunk 1]
```diff
@@ -74,6 +74,15 @@ of parallel processes to 4::
 
     SCANCODE_DEFAULT_OPTIONS=--processes 4,--timeout 120
 
+SCANCODEIO_PIPELINES_DIRS
+-------------------------
+
+This setting defines the additional locations ScanCode.io will search for pipelines.
+This should be set to a list of strings that contain full paths to your additional
```
Comment (pombredanne) [inline docs/scancodeio-settings.rst:81]:
```suggestion
This should be set to a list of comma-separated strings that contain full paths to your additional
```

File: scanpipe/apps.py [hunk 2]
```diff
@@ -83,6 +100,22 @@ def register_pipeline(self, name, cls):
...
+        """
+        Search for a Pipeline subclass in the provided file `path` and register it
+        when found.
+        """
+        module_name = path.stem
+        module = SourceFileLoader(module_name, str(path)).load_module()
+        pipeline_classes = inspect.getmembers(module, is_pipeline)
+
+        if len(pipeline_classes) > 1:
+            raise ImproperlyConfigured("Only 1 Pipeline class allowed per file.")
```
Comment (pombredanne) [inline scanpipe/apps.py:113]:
```suggestion
            raise ImproperlyConfigured(f"Only one Pipeline class allowed per pipeline file: {path}.")
```

File: scanpipe/apps.py [hunk 3]
```diff
@@ -83,6 +100,22 @@ def register_pipeline(self, name, cls):
...
+
+        elif pipeline_classes:
+            pipeline_class = pipeline_classes[0][1]
+            self.register_pipeline(name=module_name, cls=pipeline_class)
```
Comment (pombredanne) [inline scanpipe/apps.py:117]: Are we checking that the name is new and unique and does not mask the existing pipelines?

**Review by pombredanne — COMMENTED** [review]
See my comments inline.

File: scanpipe/apps.py (hunk 3 above)
Comment (tdruez) [reply scanpipe/apps.py:117]: Yes: https://github.com/nexB/scancode.io/blob/7ade8de7c0f5fee295ba7472c4811a30e06f1ab9/scanpipe/apps.py#L96-L99

**Pull request merged**